packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardTitle.base.tsx

```diff
@@ -95,72 +100,84 @@ export class DocumentCardTitleBase extends BaseComponent<IDocumentCardTitleProps
+            clientWidth,
+            needMeasurement: false
+          });
+        }
+      }
     }
+
+    return this.setState({ needMeasurement: false });
   };
 
-  private _shrinkTitle(): void {
-    if (this._doesTitleOverflow()) {
-      const { truncatedTitleFirstPiece, truncatedTitleSecondPiece } = this.state;
-      this._isTruncated = true;
+  private _shrinkTitle: () => void = () => {
+    const { truncatedTitleFirstPiece, truncatedTitleSecondPiece } = this.state;
+    if (truncatedTitleFirstPiece && truncatedTitleSecondPiece) {
+      const titleElement = this._titleElement.current;
+
+      if (!titleElement) {
+        return false;
+      }
 
-      if (truncatedTitleFirstPiece || truncatedTitleSecondPiece) {
+      if (
+        titleElement.scrollHeight > titleElement.clientHeight + TRUNCATION_VERTICAL_OVERFLOW_THRESHOLD ||
+        titleElement.scrollWidth > titleElement.clientWidth
+      ) {
         this.setState({
-          truncatedTitleFirstPiece: truncatedTitleFirstPiece!.slice(0, truncatedTitleFirstPiece!.length - 1),
-          truncatedTitleSecondPiece: truncatedTitleSecondPiece!.slice(1)
+          truncatedTitleFirstPiece: truncatedTitleFirstPiece.slice(0, truncatedTitleFirstPiece.length - 1),
+          truncatedTitleSecondPiece: truncatedTitleSecondPiece.slice(1)
         });
       }
     }
-  }
-
-  private _doesTitleOverflow(): boolean {
-    const titleElement = this._titleElement.current;
-
-    if (!titleElement) {
-      return false;
-    }
-
-    return (
-      titleElement.scrollHeight > titleElement.clientHeight + TRUNCATION_VERTICAL_OVERFLOW_THRESHOLD ||
-      titleElement.scrollWidth > titleElement.clientWidth
-    );
-  }
+  };
 
   private _updateTruncation(): void {
-    // Only update truncation if the title's size has changed since the last time we truncated
-    if (this._titleElement.current && this._titleElement.current.clientWidth !== this._truncatedTitleAtWidth) {
-      // Throttle truncation so that it doesn't happen during a window resize
-      clearTimeout(this._scrollTimerId);
-      this._scrollTimerId = this._async.setTimeout(this._startTruncation.bind(this, this.props), 250);
-    }
+    this._async.requestAnimationFrame(() => {
+      // Only update truncation if the title's size has changed since the last time we truncated
+      if (this._titleElement.current) {
+        const clientWidth: number = this._titleElement.current.clientWidth;
+        // Throttle truncation so that it doesn't happen during a window resize
+        clearTimeout(this._scrollTimerId);
+        if (this.state.clientWidth !== clientWidth) {
+          this._scrollTimerId = this._async.setTimeout(
+            () =>
+              this.setState({
+                truncatedTitleFirstPiece: undefined,
+                truncatedTitleSecondPiece: undefined,
+                needMeasurement: true
+              }),
+            250
```

Comment:
Why 250 ms? Does this mean that users will see the big title for a moment before it vanishes? It looks like this was just a hold over from the old code, but is it necessary?

Reply:
> Why 250 ms?
extends the number before
> Does this mean that users will see the big title for a moment before it vanishes?
Yes, it throttled the resize event for performance concerns in my mind.
> It looks like this was just a hold over from the old code, but is it necessary?
In my mind, throttle is necessary for performance concern.